src/pymmcore_gui/actions/widget_actions.py

```diff
@@ -89,6 +91,39 @@ class InstallDialog(QDialog, InstallWidget): ...
     return wdg
 
 
+def create_load_config(parent: QWidget) -> QDialog:
+    """Open file dialog to select a config file."""
+
+    class LoadConfigDialog(QFileDialog):
+        def exec(self) -> int:
+            (filename, _) = super().getOpenFileName(
+                None, "Select a Micro-Manager configuration file", "", "cfg(*.cfg)"
+            )
+            if filename:
+                mmc = _get_core(parent) or CMMCorePlus.instance()
+                mmc.unloadAllDevices()
```

Comment:
`mmc.loadSystemConfiguration` will itself call `mmc.unloadAllDevices` ... i don't think you need it here

Reply:
Oh, good to know. I had originally copied this code from [pymmcore-widgets](https://github.com/pymmcore-plus/pymmcore-widgets/blob/7d4f2b3e85fb39e81c10e4eeef98eef6fc5649b4/src/pymmcore_widgets/_util.py#L259) - removed it here, could update it there too...